java/server/src/org/openqa/selenium/grid/graphql/Node.java

```diff
@@ -59,6 +58,22 @@ public int getMaxSession() {
     return maxSession;
   }
 
+  public StringBuilder getCapabilities() {
+    StringBuilder str = new StringBuilder("[");
+
+    for (Map.Entry<Capabilities, Integer> entry : capabilities.entrySet()) {
+      Capabilities k = entry.getKey();
```

Comment:
Do something like the following.
```java 
List<Map> toReturn = new List<>();
...
//now inside your loop
Map<String, Object> details  = new Map<>();
details.put("browserName", entry.getKey().getBrowserName());
details.put("slot", entry.getValue());
toReturn.add(details);
```
And then have `JSON.toJSON` be called on the return from your new method which will return a `List<Map>` now.